client/src/components/Search.tsx

```diff
@@ -1,27 +1,49 @@
 import React, { useState, useRef, useEffect } from "react";
 import { SearchNodeSuggestion } from "../types/roomsearch/SearchNodeSuggestion";
+import { useRouter } from "next/router";
+import { RoomSearch } from "../logic/interfaces/RoomSearch";
 
 type Prop = {
-  roomSearcher: (searchedText: string) => SearchNodeSuggestion[];
+  roomSearcher: RoomSearch;
   onSelection: (selectedId: string) => void;
+  flag: number; //start or destination search
 }
 
-function Search({ roomSearcher, onSelection }: Prop) {
+function Search({ roomSearcher, onSelection, flag }: Prop) {
+  const router = useRouter();
   const [inputValue, setInputValue] = useState("");
   const [showDropdown, setShowDropdown] = useState(false);
   const [dropdownOptions, setDropdownOptions] = useState<SearchNodeSuggestion[]>([]);
-
   const searchRef = useRef(null);
 
-  const handleInputChange = async (event) => {
+
+  useEffect(() => {
+    if(router.isReady){
+      
+      const data = router.query;
+      if(flag==0) {
+        if(data.startNodeId as string != null) {
+          const foundSuggestion = roomSearcher.findRoomByNodeId(data.startNodeId as string);
+          setInputValue(foundSuggestion.roomName);
+        }
+      }
+      if(flag==1) {
+        if(data.endNodeId as string != null) {
+          setInputValue(roomSearcher.findRoomByNodeId(data.endNodeId as string).roomName);
+        }
+      }
+    }
+  }, [router.isReady]);
```

Comment:
The data (from router.query) will be read in the search form and passed as mentioned props inside Search component. Search shouldn't read router.query data.